### PR TITLE
Fixed inconsistency in Notification Number Box

### LINKED ISSUE #537: Notification number box showing even with no Notification
**Summary:**
The notification icon on the top-right of home screen in toolbar shows a red circle even when there is no Notification.

**Steps to reproduce:**
Close the app, and open again. (The problem resolves itself when the user taps on a given option and returns back to the home screen)

**Expected behavior:**
The red circle next to notification icon should only show-up when there is a notification.

**Observed behavior:**
The red circle next to notification icon shows-up when there is no notification on starting the app.

**Device and Android version:**

Oneplus 5T
Android 7.1.1(Nougat)
Oxygen OS

**Screenshots:**
whatsapp image 2018-01-19 at 5 46 06 pm

### PR DESCRIPTION
**Pull request by miPlodder**
Fixes #537 

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by miPlodder** [comment]
@therajanmaurya @tarun0 @dilpreet96 Please review this PR.

**Comment by dilpreet2028** [comment]
@miPlodder Good Catch!
But also add `getMvpView().showNotificationCount(0);`  to this line: https://github.com/openMF/mifos-mobile/blob/development/app/src/main/java/org/mifos/mobilebanking/presenters/HomeOldPresenter.java#L195

**Comment by therajanmaurya** [comment]
@miPlodder Please open the new activity on click notification button instead of fragment transaction in home activity.

**Comment by miPlodder** [comment]
@dilpreet96 @therajanmaurya Sorry, for delayed reply (had some issues with my system). I will make the required changes and send a PR.

**Comment by miPlodder** [comment]
@therajanmaurya @dilpreet96 I have made the request changes and updated the PR. Please review the PR.

**Screenshot of Notification Activity** 

( I have added back button for improving User Experience )

**Comment by miPlodder** [comment]
@therajanmaurya @dilpreet96 I would like to add _**Toast message**_ in the Notification Activity, when there is no Notification for the User. (It will improve the **UX** and will be better than showing an empty recycler view).

If you think that it's a valuable addition, I would like to work on it.

**Comment by dilpreet2028** [comment]
@miPlodder , Instead of adding a toast you can show a text in the centre saying `No Notifications`

File: app/build.gradle [hunk 1]
```diff
@@ -66,6 +67,7 @@ dependencies {
 
     implementation fileTree(dir: 'libs', include: ['*.jar'])
 
+    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
```
Comment (dilpreet2028) [inline app/build.gradle]: Are you using constraint layout anywhere? if no then remove it

File: app/build.gradle [hunk 2]
```diff
@@ -81,7 +83,7 @@ dependencies {
     implementation "com.google.android.gms:play-services-oss-licenses:$rootProject.playServicesVersion"
 
     //Square dependencies
-    implementation ('com.squareup.retrofit2:retrofit:$rootProject.retrofitVersion') {
+    implementation('com.squareup.retrofit2:retrofit:$rootProject.retrofitVersion') {
```
Comment (dilpreet2028) [inline app/build.gradle]: Add the space which you have removed

File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/HomeOldFragment.java [hunk 3]
```diff
@@ -132,11 +133,16 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
...
         View count = menuItem.getActionView();
         tvNotificationCount = (TextView) count.findViewById(R.id.tv_notification_indicator);
+        presenter.getUnreadNotificationsCount();
         count.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                ((BaseActivity) getActivity()).replaceFragment(NotificationFragment.newInstance(),
-                        true, R.id.container);
+
+                startActivity(
```
Comment (dilpreet2028) [inline app/src/main/java/org/mifos/mobilebanking/ui/fragments/HomeOldFragment.java]: Please change this into 1 line or 2, no need of intending it in this way

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -1,10 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
```
Comment (dilpreet2028) [inline app/src/main/AndroidManifest.xml]: @miPlodder , why you have made these changes in manifest, please revert them and just add a new <activity> tag for the activity that you have created.

**Review by dilpreet2028 — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (miPlodder) [reply app/build.gradle]: Ok (Added by Studio when I created new activity for Notification)

File: app/build.gradle (hunk 2 above)
Comment (miPlodder) [reply app/build.gradle]: Ok

File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/HomeOldFragment.java (hunk 3 above)
Comment (miPlodder) [reply app/src/main/java/org/mifos/mobilebanking/ui/fragments/HomeOldFragment.java]: Ok

File: app/src/main/AndroidManifest.xml (hunk 4 above)
Comment (miPlodder) [reply app/src/main/AndroidManifest.xml]: Ok

File: app/src/main/res/layout/fragment_notification.xml [hunk 5]
```diff
@@ -8,4 +8,13 @@
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         android:id="@+id/rv_notifications"/>
+    <TextView
+        android:id="@+id/tv_no_notification_msg"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:gravity="center"
+        android:visibility="gone"
+        android:text="No Notifications"
```
Comment (dilpreet2028) [inline app/src/main/res/layout/fragment_notification.xml]: @miPlodder , Don't use raw strings instead create a resource in strings.xml for this.

**Review by dilpreet2028 — CHANGES_REQUESTED** [review]
@miPlodder also post a Screenshot of notifications screen.

File: app/src/main/res/layout/fragment_notification.xml (hunk 5 above)
Comment (miPlodder) [reply app/src/main/res/layout/fragment_notification.xml]: I will make the changes and update the PR.

**Comment by miPlodder** [comment]
When activity has no notification,

_old 1

When activity has notification,

_old 2

**Comment by miPlodder** [comment]
@dilpreet96 
I would like to add SwipeRefreshLayout to this activity to make activity consistent with other activities and it will surely enhance the UI/UX for the end user.

Should I add it in this PR ?

**Comment by dilpreet2028** [comment]
Yeah sure, go ahead

**Review by dilpreet2028 — APPROVED** [review]

File: app/src/main/res/layout/action_bar_notification_layout.xml [hunk 6]
```diff
@@ -25,6 +25,7 @@
         android:textColor="@color/white"
         android:layout_height="wrap_content"
         android:gravity="top|end"
+        android:visibility="gone"
```
Comment (miPlodder) [inline app/src/main/res/layout/action_bar_notification_layout.xml:28]: I have made a change here, this **TextView** was being show for a fraction of second, until we make it's visibility to **GONE** using Java 
(https://github.com/miPlodder/mifos-mobile/blob/7c93d583dbc8482a111943bdf64b32284a5194d6/app/src/main/java/org/mifos/mobilebanking/ui/fragments/HomeOldFragment.java#L289)

**Comment by miPlodder** [comment]
@dilpreet96 I have updated the PR. SwipeRefreshLayout is added and made some changes in TextView as mentioned in the comment above.

Please review this PR.

Screenshot : 

swipe_pr

**Comment by dilpreet2028** [comment]
Good work @miPlodder

**Pull request merged**